docs/monorepo.md

```diff
@@ -133,22 +133,70 @@ Note that if you're building with Webpack, you may need to turn off [`resolve.sy
+
+Create PRs with the necessary changes and merge them to `master` before publishing. Lerna will add a `gitHead` field to each published package's `package.json`. That field contains the hash of the Git commit that the package was published from. It's better if this commit hash is a permanent one from the `master` branch, rather than an ephemeral commit from a local branch.
+
+### Checkout the latest master locally and build the packages
+
+Always publish from the latest `master` branch, so that the package contents come from a verified source that everyone has access to. It's too easy to publish a NPM package from a local branch, or even uncommitted local modifications that are invisible to anyone but you.
+
+Build the `dist/` directories (the transpiled package content that will be published) from scratch.
+
+```
+git checkout master
+git pull
+git status (should be clean!)
+npm run distclean
+npm ci
+npm run build-packages
+```
+
+### Publishing all outdated packages
+
+It might be good to start un-authenticated, since Lerna doesn't have `--dry-run` option like NPM does: `npm logout`.
+
+Now run: `npx lerna publish from-package`
+
+Lerna will show a list of packages that have versions higher than the latest one published in the NPM registry. Verify that this is indeed the list of packages that you want to publish. If something looks off, abort!
+
+If you say "yes" to the Lerna prompt, and are not authenticated, the publishing will fail with authentication error. Better to say "no".
+
+Now make sure you're logged in at this point, we're going to publish 🚀: `npm whoami`, `npm login`. Enter your username, password, and the OTP code.
+
+Craft the following command, we'll add `--yes` to skip prompts and save OTP cycle time. `--dist-tag next` is optional, use it when publishing unstable versions.
+
+Wait for your NPM OTP (one time password) cycle to start -- if your NPM account is correctly set up for maximum security, you'll need to enter the OTP code when publishing, even though you've already logged in with another OTP. Write the code into the command and publish:
+
+```
+NPM_CONFIG_OTP=[YOUR_OTP_CODE] npx lerna publish --dist-tag next from-package --yes
+```
+
+Pat yourself on the back, you published!
+
+### Publishing a single package
+
+What if you want to publish just one updated package? `lerna publish from-package` either publishes all eligible packages, or nothing. It doesn't give you a choice. That's where you need `lerna publish from-git`.
+
+To publish only selected packages, you need to create Git tags in form `name@version`. For example:
+```
+git tag "@automattic/calypso-build@6.1.0"
+```
+or
+```
+git tag "@automattic/components@1.0.0"
+```
+
+Now `lerna publish from-git` will offer to publish only the packages that have a matching Git tag on the current `HEAD` revision.
+
+The rest of the workflow is exactly the same as in the `from-package` case.
```

Comment:
Do we want to suggest `from-git`? It seemed very problematic when we tried to use it early on. Maybe we should suggest adding `private:true` to `package.json` temporarily.

Reply:
Seems like it's the most straightforward way to publish a single package. The `private:true` one seems even more like a hack 😉 What was problematic about it before?

Reply:
@sirreal I use `from-git` every time I'm publishing, haven't run into any issues so far, and I can recommend this workflow 🙂 If there are any risks, let's describe them.